server/src/main/java/org/elasticsearch/action/explain/ExplainResponse.java

```diff
@@ -129,4 +157,92 @@ public void writeTo(StreamOutput out) throws IOException {
             getResult.writeTo(out);
         }
     }
+
+    private static final ConstructingObjectParser<ExplainResponse, Void> PARSER = new ConstructingObjectParser<>("explain", true,
+        arg -> new ExplainResponse((String) arg[0], (String) arg[1], (String) arg[2], (boolean) arg[3], (Explanation) arg[4],
+            (GetResult) arg[5]));
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), _INDEX);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), _TYPE);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), _ID);
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), MATCHED);
+        final ConstructingObjectParser<Explanation, Void> explanationParser = new ConstructingObjectParser<>("explanation", true,
+            arg -> {
+                if ((float) arg[0] > 0) {
+                    return Explanation.match((float) arg[0], (String) arg[1], (Collection<Explanation>) arg[2]);
+                } else {
+                    return Explanation.noMatch((String) arg[1], (Collection<Explanation>) arg[2]);
+                }
+            });
+        explanationParser.declareFloat(ConstructingObjectParser.constructorArg(), VALUE);
+        explanationParser.declareString(ConstructingObjectParser.constructorArg(), DESCRIPTION);
+        explanationParser.declareObjectArray(ConstructingObjectParser.constructorArg(), explanationParser, DETAILS);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), explanationParser, EXPLANATION);
+        PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> GetResult.fromXContentEmbedded(p), GET);
```

Comment:
does this parse the explanation recursively like it's printed out in toXcontent?